src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -500,10 +499,16 @@ public Boolean isStockIdExists(final List<String> stockIds) {
 
 		final List<TransactionReportRow> transactions = new ArrayList<>();
 		try {
-			final String sql = "SELECT i.userid,i.lotid,i.trndate,i.trnstat,i.trnqty,i.sourceid,l.listname, i.comments,"
-					+ "(CASE WHEN trnstat = 1 AND trnqty >= 0 THEN '" + TransactionType.DEPOSIT.getValue()
-					+ "' WHEN trnstat = 0 AND trnqty < 0 THEN '" + TransactionType.RESERVATION.getValue()
-					+ "' WHEN trnstat = 1 AND trnqty < 0 THEN '" + TransactionType.WITHDRAWAL.getValue()
+			final String sql = "SELECT i.userid,i.lotid,i.trndate,"
+					+ "(CASE WHEN trnstat = 0 THEN '" + TransactionStatus.PENDING.getValue()
+					+ "' WHEN trnstat = 1 THEN '" + TransactionStatus.CONFIRMED.getValue()
+					+ "' WHEN trnstat = 2 THEN '" + TransactionStatus.CANCELLED.getValue()
+					+ "' END) as trnStatus, "
+					+ " i.trnqty,i.sourceid,l.listname, i.comments,"
+					+ "(CASE WHEN trntype = 0 THEN '" + TransactionType.DEPOSIT.getValue()
```

Comment:
Fix this case, 4 for deposit, use ids in enum instead of hardcoded values

Reply:
Done